### PR TITLE
all: Hide implementations behind interfaces for mocked testing

### PR DESCRIPTION
**Pull request by imsodin**
As per https://github.com/syncthing/syncthing/pull/5529#discussion_r257988426 here comes the PR from hell: Remove the `...Intf` interfaces in gui.go in favor of making `config.Wrapper`, `model.Model` and `connections.Service` interfaces.

I tried to order the inteface methods thematically, please tell me if it's not helpful.

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
This looks actually sane. Need a bigger screen tho.

**Comment by imsodin** [comment]
@AudriusButkevicius Got a moment with a bigger screen yet?

**Comment by AudriusButkevicius** [comment]
If only GitHub had a remind me bot. @remindme in 6hrs

File: lib/locations/locations.go [hunk 1]
```diff
@@ -49,7 +49,6 @@ func init() {
 		panic(err)
 	}
 }
-
```
Comment (AudriusButkevicius) [inline lib/locations/locations.go]: why?

Comment (imsodin) [reply lib/locations/locations.go]: Don't know, not to check your (quite impressive) attentiveness.

**Comment by AudriusButkevicius** [comment]
This is an awesome change.

**Pull request merged**